tools/chpl-language-server/test/test_server.py

```diff
@@ -0,0 +1,611 @@
+
+        assert len(client.diagnostics) == 0
+
+
+@pytest.mark.asyncio
+async def test_go_to_definition_use_standard(client: LanguageClient):
+    """
+    Ensure that go-to-definition works on standard module symbols
+    that are used or imported.
+    """
+
+    file = """
+           use IO;
+           use List, Map;
+           import Time;
+           """
+
+    mod_IO = standard_module("standard/IO.chpl")
+    mod_List = standard_module("standard/List.chpl")
+    mod_Map = standard_module("standard/Map.chpl")
+    mod_Time = standard_module("standard/Time.chpl")
+
+    with source_file(client, file) as doc:
+        await check_goto_decl_def(client, doc, pos((0, 4)), mod_IO)
+        await check_goto_decl_def(client, doc, pos((1, 4)), mod_List)
+        await check_goto_decl_def(client, doc, pos((1, 10)), mod_Map)
+        await check_goto_decl_def(client, doc, pos((2, 8)), mod_Time)
+
+        assert len(client.diagnostics) == 0
+
+
+@pytest.mark.asyncio
+async def test_go_to_definition_standard_rename(client: LanguageClient):
+    """
+    Ensure that go-to-definition works with using/importing symbols,
+    even if renaking is in use.
+    """
+
+    file = """
+           use IO as OI;
+           import IO.{ioMode as im};
+           use List only list;
+           """
+
+    mod_IO = standard_module("standard/IO.chpl")
+    mod_List = standard_module("standard/List.chpl")
+
+    with source_file(client, file) as doc:
+        await check_goto_decl_def(
+            client, doc, pos((0, 4)), mod_IO, expect_str="module IO"
+        )
+        await check_goto_decl_def(
+            client, doc, pos((0, 10)), mod_IO, expect_str="module IO"
+        )
+        await check_goto_decl_def(
+            client, doc, pos((1, 7)), mod_IO, expect_str="module IO"
+        )
+        await check_goto_decl_def(
+            client, doc, pos((1, 11)), mod_IO, expect_str="enum ioMode"
+        )
+        await check_goto_decl_def(
+            client, doc, pos((1, 21)), mod_IO, expect_str="enum ioMode"
+        )
+        await check_goto_decl_def(
+            client, doc, pos((2, 4)), mod_List, expect_str="module List"
+        )
+        await check_goto_decl_def(
+            client, doc, pos((2, 14)), mod_List, expect_str="record list"
+        )
+
+        assert len(client.diagnostics) == 0
+
+
+@pytest.mark.asyncio
+async def test_go_to_record_def(client: LanguageClient):
+    """
+    Ensure that 'go to definition' on a type actually works.
+    """
+
+    file = """
+           record myRec {}
+           var x: myRec;
+           var y = new myRec();
+           """
+
+    with source_file(client, file) as doc:
+        await check_goto_decl_def(client, doc, pos((0, 7)), pos((0, 7)))
+        await check_goto_decl_def(client, doc, pos((1, 7)), pos((0, 7)))
+        await check_goto_decl_def(client, doc, pos((2, 12)), pos((0, 7)))
+
+        assert len(client.diagnostics) == 0
```

Comment:
I think you should test `var z: string` as well since it seems most prone to breakage, even when other goto typedef is not :smile: